src/main/scala/mastermind/Mastermind.scala

```diff
@@ -13,19 +13,48 @@ val validColors = List('B', 'Y', 'R', 'G')
 
 /** Get a random color from the list of valid colors */
 def getRandomColor(): Color =
-  ???
+  val rand = new scala.util.Random
+  val ind :Int = rand.nextInt(validColors.length)
+  validColors.apply(ind)
+
 
 /** Given four colors, make a board from them */
 def makeBoardFromColors(c1: Color, c2: Color, c3: Color, c4: Color): Board =
-  ???
+  c1.toString() + c2.toString() + c3.toString() + c4.toString()
 
 /** Create a random board */
 def getRandomBoard(): Board =
-  ???
+  val c1 = getRandomColor()
+  val c2 = getRandomColor()
+  val c3 = getRandomColor()
+  val c4 = getRandomColor()
+  makeBoardFromColors(c1, c2, c3, c4)
 
 /** Play one round of the game */
 def playRound(board: Board): (Int, Int) =
-  ???
+  var correct_place = 0
+  var correct_val = 0
+  println("Enter a guess for spot 1:")
+  val spot_one = readLine().toString()
+  println("Enter a guess for spot 2:")
+  val spot_two = readLine().toString()
+  println("Enter a guess for spot 3:")
+  val spot_three = readLine().toString()
+  println("Enter a guess for spot 4:")
+  val spot_four = readLine().toString()
+  val guess = spot_one + spot_two + spot_three + spot_four
+  scoreGuess(board, guess)
```

Comment:
Great readable code here, I tried doing my solutions in shorter approaches which can look like less code but yours are very clear! I think it would have been better for me to take an approach like this since it's much easier to go back and read than what I wrote.